lambda_code/transfer/User/index.py

```diff
@@ -0,0 +1,67 @@
+from cfn_custom_resource import CloudFormationCustomResource
+from _metadata import CUSTOM_RESOURCE_NAME
+
+
+class User(CloudFormationCustomResource):
+    """
+    Properties:
+        Role: str: role for user, should include permissions to access a bucket
+        Policy: str: policy for limiting access of user
+        ServerId: str: the transfer server id
+        SshPublicKeyBody: str: base64 encoded public key for user
+        UserName: str: name of user
+        HomeDirectory: str: home directory folder in bucket
+    """
+    RESOURCE_TYPE_SPEC = CUSTOM_RESOURCE_NAME
+    DISABLE_PHYSICAL_RESOURCE_ID_GENERATION = True
+
+    def validate(self):
+        self.role = self.resource_properties.get('Role')
```

Comment:
Since these are required, I'd do `self.resource_properties['Role']`, and let it raise if the key does not exist.

Same for the other required values